### PR TITLE
Feature/callback queue options

### PR DESCRIPTION
**Pull request by jabou**

### CODE REVIEW

File: Versioner/PrinceOfVersions.swift [hunk 1]
```diff
@@ -21,6 +21,14 @@ public class PrinceOfVersions: NSObject {
         public let result: Result<UpdateInfo, UpdateInfoError>
     }
 
+    /// The queue on which the completion handler is dispatched
+    ///
+    /// By default, OperationQueue of the URLSession's delegateQueue is set to default one (nil)
+    @objc public enum CallbackQueue: Int {
```
Comment (Truba) [inline Versioner/PrinceOfVersions.swift]: Why not just use DispatchQueue, and let user set preferred one, similar to what Alamofire does?

File: Versioner/PrinceOfVersions.swift [hunk 2]
```diff
@@ -183,6 +204,16 @@ private extension PrinceOfVersions {
         return publicKey
     }
 
+     static func _dispatch(block: @escaping (() -> Void), on queue: CallbackQueue) {
```
Comment (Truba) [inline Versioner/PrinceOfVersions.swift]: e tu, mi fili brute?

File: Versioner/PrinceOfVersions.swift [hunk 3]
```diff
@@ -127,21 +140,29 @@ public extension PrinceOfVersions {
...
                 if let minimumSdk = info.minimumSdkForLatestVersion, minimumSdk > info.sdkVersion {
-                    noNewVersion(info.isMinimumVersionSatisfied, info.metadata)
+                    PrinceOfVersions._dispatch(block: {
+                        noNewVersion(info.isMinimumVersionSatisfied, info.metadata)
+                    }, on: callbackQueue)
                 } else {
                     let latestVersion = info.latestVersion
                     if (latestVersion > info.installedVersion) && (!latestVersion.wasNotified || info.notificationType == .always) {
-                        newVersion(latestVersion, info.isMinimumVersionSatisfied, info.metadata)
+                        PrinceOfVersions._dispatch(block: {
```
Comment (Truba) [inline Versioner/PrinceOfVersions.swift]: I would go with `callbackQueue.aysinc` like in DispatchQueue 😄

File: Versioner/PrinceOfVersions.swift (hunk 2 above)
Comment (jabou) [reply Versioner/PrinceOfVersions.swift]: image

🙊

File: Versioner/PrinceOfVersions.swift (hunk 3 above)
Comment (jabou) [reply Versioner/PrinceOfVersions.swift]: 👍 yep, much better :)

File: Versioner/PrinceOfVersions.swift (hunk 1 above)
Comment (jabou) [reply Versioner/PrinceOfVersions.swift]: Changed to `DispatchQueue`

**Pull request merged**